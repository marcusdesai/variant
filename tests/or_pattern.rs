use variant::try_variant;

struct Test {
    a: u8,
    b: Option<u8>,
}

fn main() {
    let val = Test { a: 1, b: Some(2) };
    let _ = try_variant!(val, Test { a, b: Some(1) | Some(2) });
}
