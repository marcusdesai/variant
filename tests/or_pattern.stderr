error: `variant` cannot match `or` patterns
  --> tests/or_pattern.rs:10:44
   |
10 |     let _ = try_variant!(val, Test { a, b: Some(1) | Some(2) });
   |                                            ^^^^^^^^^^^^^^^^^
